### DISCUSSION TITLE
client certificate expiration expired - how to extend

### DISCUSSION THREAD

**Discussion by un99known99** [pivpn/pivpn#1707] [Q&A]
Hi,
I checked for the possibility to extend a client cert generated with “pivpn -a”.
As of now, I only found a way to extend a server cert.

So how can a client cert be extended, or do I have to generate a new one?

**Comment by FLUPS** [comment]
You need to generate a new one, revoke the old previous

un99known99 ***@***.***> schrieb am Di., 21. März 2023, 09:05:

**Reply by un99known99** [reply]
expired cant revoked ....
how can they be deleted to create a new one with same name?

**Reply by coolapso** [reply]
you don't. 
If its expired, it won't be accepted and access with that certificate will be denied. 
just create another profile with the same name.

**Reply by un99known99** [reply]
thx for the info, how can the ones showing up via pivpn -l as EXPIRED be **_deleted_**???

**Reply by coolapso** [reply]
not possible

https://docs.pivpn.io

**Reply by un99known99** [reply]
thx a lot for ur feedback
